js/lib/beautify.js

```diff
@@ -1673,13 +1674,21 @@
                 var allow_e = true;
                 var local_digit = digit;
 
-                if (c === '0' && parser_pos < input_length && /[Xxo]/.test(input.charAt(parser_pos))) {
-                    // switch to hex/oct number, no decimal or e, just hex/oct digits
+                if (c === '0' && parser_pos < input_length && /[Xxob]/.test(input.charAt(parser_pos))) {
+                    // switch to hex/oct/bin number, no decimal or e, just hex/oct/bin digits
                     allow_decimal = false;
                     allow_e = false;
                     c += input.charAt(parser_pos);
                     parser_pos += 1;
-                    local_digit = /[o]/.test(input.charAt(parser_pos)) ? digit_oct : digit_hex;
+                    if ( /[b]/.test(input.charAt(parser_pos)) ) {
+                        local_digit = digit_bin;
+                    } else {
```

Comment:
This project uses `} else if (...) {` where reasonable and appropriate, this is one of those cases. 

We attempt to maintain parity between python and javascript versions of the code.  Not always possible, but where you see `elif ...:` in python, you should see `} else if (...) {` in javascript.